### ISSUE TITLE
Student: submit responses: warn if there are unanswered questions

### ISSUE THREAD

**Issue by Haozhe321** [TEAMMATES/teammates#8598]
When students click on the submit button, they will receive this confirmation message below 
__________________________________________________________________________________________________
![image](https://user-images.githubusercontent.com/26651556/37037821-21fc4742-218e-11e8-8a5b-c0a8efc7b4ba.png)
__________________________________________________________________________________________________
This message is shown regardless of whether they finished all the questions. If a feedback session has many questions, it is possible for them to miss one question or two. The confirmation popup modal above should let them know that they have unanswered questions. 

If they left those questions out on purpose that's fine, but this message is helpful for those that unintentionally missed a question or two.

**Comment by craaaa** [comment]
This sounds like a good idea, but I wonder about how this would work for optional questions. For instance, since `Essay Questions` only specify a maximum number of people that a respondent can give feedback to, would giving less than the maximum amount of feedback count as 'missing out' on a question? (Playing the devil's advocate here; I think checking that submissions are complete would be a good feature.)

**Comment by Haozhe321** [comment]
That's a very good point. In the particular case of `essay questions`, I think as long as the student has answered the question(not necessarily exhausted all the choices), the question should be regarded as answered. It's not a requirement of the question for them to hit the maximum number. I think if students navigate away from the question, it generally means they are done with it. 
 
To anyone else reading, here is the question we are discussion: 
![image](https://user-images.githubusercontent.com/26651556/37043123-d657da38-219a-11e8-9b20-3a4f1b11c826.png)

**Comment by Haozhe321** [comment]
For other types of questions, I think the above principle can work as well. i.e. for MSQ, as long as they have chosen at least one choice, the question is regarded as answered. The only exception are questions which require selecting a **minimum** number of choices. In this case, the UI already prevents them from submitting like here: 
![image](https://user-images.githubusercontent.com/26651556/37043724-654e03ba-219c-11e8-9704-88a8a49fcd0f.png)
(Strangely, they can submit these questions without answering at all; but if they do choose to answer and select less than the minimum number, submission is denied.)

**TLDR**: if they leave any question totally blank, we should remind them that they have unanswered question.

**Comment by sukanta-27** [comment]
@Haozhe321 I really like this idea, (This came to my mind while I was working on #7993 ). 

>Strangely, they can submit these questions without answering at all; but if they do choose to answer and select less than the minimum number, submission is denied.)

Not strangely, it is because partial answers can be submitted. The modal text is notifying the users that the questions they have answered are saved. (You can skip questions for now, but if you do decide to answer, answer it the right way :smile: )

By the way, I am wondering what will be the optimal way to implement this? Are you suggesting a modal or can a status message do the job?

**Comment by damithc** [comment]
Perhaps show a summary of questions in the status message as a visual feedback indicating which questions are answered and which are not? I remember @tshradheya also mentioned something like this with respect to the email confirmation sent to the student after a submission.

**Comment by sukanta-27** [comment]
@damithc sir, I guess this is also related to the "Upgrade" you discussed in #7993 

>Perhaps show a summary of questions in the status message as a visual feedback indicating which questions are answered and which are not

About this, how about we show just how many questions are answered and how many are unanswered, instead of the full summary? Given the fact that after #8367 merged we will already have one status message, and after successful submission there will be one too. And with that if we give so much information in another status message, wouldn't it be a lot of status messages in one page? :sweat_smile: 

Also I second the status message decision, it is better than modal IMO. But I will suggest we do this after #8367 gets merged, cause currently our status message checking mechanism is a little flawed, does not take into account multiple message, I have worked on the testability of status messages in that PR, after that adding a status message should be easier. Otherwise anyone who will work on it will have to reinvent the wheel.

**Comment by Haozhe321** [comment]
My inclination is to use the same modal that they see after clicking on the submit button(the first image in this issue). If they have not completed all the questions, add a paragraph below telling them so(also informing which questions are left empty, just like certain online registration forms). It looks simple and achieves it's goal.

**Comment by jacoblipech** [comment]
Hi @Haozhe321, are you working on this issue? If not can I do it? :)

**Comment by Haozhe321** [comment]
@jacoblipech for new contributors, we recommend taking a `d.FirstTimer` issue first before going into the `d.Contributors`. You may want to revisit this after doing a first timer issue :)

**Comment by jacoblipech** [comment]
okay thank you @Haozhe321 I am working on this!

**Comment by jacoblipech** [comment]
Hi @Haozhe321, sorry for the delay, I have added the incomplete questions message upon pressing the submit button. For the questions that have any empty messages, I will add the question index to an array and eventually, it will show the user all the incompleted questions in a string. It will look something like the image below:
<img width="960" alt="screen shot 2018-04-05 at 10 49 57 am" src="https://user-images.githubusercontent.com/28522090/38345019-54e66c8a-38bf-11e8-821b-55fcf24dfd34.png">

Since the users can save partial answers, I believe this function should mainly inform users of any incomplete questions before they leave the page. Currently, all questions are accounted as I did not set any questions that must be filed up or ignored.

Do let me know what you think before I move on to the testing and make a PR :)

**Comment by jacoblipech** [comment]
@Haozhe321 Thanks for the prompt reply. Right now, on top of the message shown, there is also a separate modal showing the incomplete questions message after the successful modal.

Personally, I think it looks good if I show the incomplete questions line together with the successful modal and add it after the "All responses have been successfully recorded" line. 

![screen shot 2018-04-05 at 2 08 14 pm](https://user-images.githubusercontent.com/28522090/38349715-ea9f7576-38da-11e8-82d6-a4dd18b75072.png)

**Comment by jacoblipech** [comment]
>I prefer to have it on the modal. We may have to use a ❗️ in front of that sentence to attract attention. 

Okay, I will work towards this style.
